### DISCUSSION TITLE
Lifting PyMuPDF's version pin from requirements.txt

### DISCUSSION THREAD

**Discussion by PythonTryHard** [PaddlePaddle/PaddleOCR#8150] [Ideas & Features]
Considering PyMuPDF is [only used to open PDFs](https://github.com/PaddlePaddle/PaddleOCR/search?q=fitz), is it necessary to hard pin the dependency to `1.19.0` as seen [here](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppstructure/recovery/requirements.txt) and [here](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/requirements.txt)? 

In addition, should `README.md` and `Quickstart.md` add the steps to troubleshoot PyMuPDF's installation should pre-built wheels be unavailable? By hard-pinning, sooner or later the troubleshooting for PyMuPDF is gonna come as people move to newer Python versions, and 1.19.0 not supporting pre-built wheels past 3.10 means as Python moves forwards, more people will complain about `pip install` error as the average user expect a plug-and-play experience of pre-built wheels (as per the spirit of the Quickstart's effectively 3 lines to start OCRing), rather having to download compile tools and compile from scratch or install PyMuPDF; and that is assuming people actually spend time to read the error log and understanding it, with "people" referring to "the average user", not "someone deep into Python".

@an1018 as the one that authored 1a9926a7fac52cfdfcac2e07ec65f1e30664153d introducing PyMuPDF as dependency. Yes, dependency pinning is good software writing techniques, but considering its use in code, what's your opinion? Do we:
- Lift the `==` completely (i.e., `PyMuPDF`)?
- Pin against major version bump (`PyMuPDF="^1.20.0"` or `PyMuPDF<2.0.0`)?
- Pin against minor version bumps (`PyMuPDF="~1.20.0"` or `PyMuPDF<1.21.0`)?

while taking into account that PyMuPDF 1.21.0's pre-built wheel for 3.11 is [in](https://github.com/pymupdf/PyMuPDF/issues/1993#issuecomment-1292761047) [consideration](https://github.com/pymupdf/PyMuPDF/issues/1992#issuecomment-1294049693) [upstream](https://github.com/pymupdf/PyMuPDF/issues/1993#issuecomment-1293582544)?

**Comment by an1018** [comment]
Thanks for your sincere suggestion. We will modify it later.

**Reply by PythonTryHard** [reply]
That would be lovely. Thank you for your consideration.
